test/suite/test_prepare27.py

```diff
@@ -0,0 +1,110 @@
+# Test that a prepared update that has been aborted is not selected as the base value.
+class test_prepare27(wttest.WiredTigerTestCase):
+    format_values = [
+        ('column', dict(key_format='r', value_format='S')),
+        ('column-fix', dict(key_format='r', value_format='8t')),
+        ('integer-row', dict(key_format='i', value_format='S')),
+        ('string-row', dict(key_format='S', value_format='S')),
+    ]
+    scenarios = make_scenarios(format_values)
+
+    def create_key(self, i):
+        return str(i) if self.key_format == 'S' else i
+
+    def create_value(self, i):
+        return str(i) if self.value_format == 'S' else i
+
+    def evict_cursor(self, uri, session_cfg, nrows):
+        s = self.conn.open_session()
+        s.begin_transaction(session_cfg)
+        evict_cursor = s.open_cursor(uri, None, "debug=(release_evict)")
+        for i in range(1, nrows + 1):
+            evict_cursor.set_key(self.create_key(i))
+            evict_cursor.search()
+            evict_cursor.reset()
+        s.rollback_transaction()
+        evict_cursor.close()
+
+    def test_prepare33(self):
+        uri = 'table:test_prepare27'
+        create_params = 'key_format={},value_format={}'.format(self.key_format, self.value_format)
+        self.session.create(uri, create_params)
+
+        # Apply a series of updates to a key.
+        num_updates = 5
+        key = self.create_key(1)
+
+        cursor = self.session.open_cursor(uri)
+        for ts in range(1, num_updates + 1):
+            with self.transaction(commit_timestamp = ts):
+                value = ts
+                cursor[key] = self.create_value(value)
+        
+        # Set the stable timestamp.
+        self.conn.set_timestamp('stable_timestamp=' + self.timestamp_str(2))
+    
+        # At this point we have 5 updates associated with a key in the update chain:
+        # 1 -> 2 -> 3 -> 4 -> 5
+
+        # By evicting, the latest update is moved to the DS and the rest to the HS.
+        self.evict_cursor(uri, None, 1)
+
+        # Perform a new update through a prepared transaction and evict the page.
+        num_updates = num_updates + 1
+        ts = num_updates
+
+        self.session.begin_transaction()
+        cursor[key] = self.create_value(ts)
+        self.session.prepare_transaction('prepare_timestamp=' + str(ts))
+
+        # Note that we need to configure the session to ignore prepared updates to be able to
+        # perform eviction. After eviction, we should have the prepared update on the DS and the
+        # rest in the HS.
+        self.evict_cursor(uri, "ignore_prepare=true", 1)
+
+        # Roll back the prepared transaction. This will mark the prepared update as aborted and
+        # bring back the previous update to the update chain.
+        self.session.rollback_transaction()
+
+        # By calling RTS, this should bring the latest stable update to the update chain and leave
+        # the other stables ones in the HS.
```

Comment:
Explaining what the update chain looks like after RTS will be helpful to understand more of this issue more.

Reply:
Updated in e58c53a750779b4be6dacf3fc37890f95d190412, let me know if that's good enough.
Can you explain why 6 is before 5? Shouldn't it be 2 -> 5 -> 6?

Reply:
To get more context on why we have an update chain like 6-> 5. As part of the prepared transaction rollback, we need to append the last update from the history store to the update chain to leave in the state of `Datastore update (6)(aborted) ` -> `latest history store update (5)`.